test/cmd/legacy-script.sh

```diff
@@ -350,6 +350,17 @@ runTests() {
     kubectl config set-context "${CONTEXT}" --namespace="${ns_name}"
   }
 
+  # Generate a namespace with restricted policy
+  create_and_use_new_restricted_namespace() {
```

Comment:
This is the common place for the tests can use, I think we don't need such a specific function in here. We can label namespace restricted directly in debug tests.

Reply:
That's good!
But, how can I get the namespace name in other function generated by `create_and_use_new_namespace()`?
Change `ns_name` local to global?

https://github.com/kubernetes/kubernetes/pull/117543/files/0986936911928fb1304faab5dd1c52db372431a8#diff-c941384878c552d1e092ddd897e915a7da98773f523ae7f14201b084b1a4d1dbR346

Reply:
This will be set as default namespace, maybe you can get it via `kubectl get namespace`?

Reply:
I decided to get the namespace name from current-context as below.

```console
$ kubectl config view --minify --output 'jsonpath={.contexts[?(@.name == "'$(kubectl config current-context)'")].context.namespace}'
namespace-1684214689-928
```

Reply:
There should be simpler way, I'd try `kubectl get namespace`

Reply:
kubectl Integration Test (`make test-cmd`) creates many namespaces in each test case.
This is the result of running `kubectl get namespace` within our test case for debugging profile.
I'm very sorry, but I don't know how to determine the current namespace from `kubectl get namespace`...
Is it possible?

```console
NAME                         STATUS   AGE
default                      Active   7m4s
kube-node-lease              Active   3m4s
kube-public                  Active   7m4s
kube-system                  Active   7m4s
namespace-1684217800-12369   Active   3m16s
namespace-1684217807-19631   Active   3m9s
namespace-1684217814-23836   Active   3m2s
namespace-1684217815-10612   Active   3m1s
namespace-1684217820-14583   Active   2m55s
namespace-1684217827-15629   Active   2m48s
namespace-1684217828-32747   Active   2m48s
namespace-1684217834-24517   Active   2m42s
namespace-1684217835-12658   Active   2m41s
namespace-1684217838-28363   Active   2m38s
namespace-1684217847-16571   Active   2m29s
namespace-1684217861-13779   Active   2m15s
namespace-1684217868-7738    Active   2m8s
namespace-1684217869-24980   Active   2m7s
namespace-1684217871-23451   Active   2m5s
namespace-1684217871-25734   Active   2m5s
namespace-1684217881-9876    Active   115s
namespace-1684217883-30526   Active   113s
namespace-1684217892-9408    Active   104s
namespace-1684217894-20121   Active   102s
namespace-1684217914-20208   Active   82s
namespace-1684217920-442     Active   76s
namespace-1684217958-3975    Active   38s
namespace-1684217992-21847   Active   4s
namespace-1684217994-12988   Active   2s
namespace-1684217995-6740    Active   1s
namespace-1684217996-3114    Active   0s
some-other-random            Active   104s
```

Reply:
Ah, sorry, my mistake. What about creating a custom namespace with restricted(basically moving what you did in `create_and_use_new_restricted_namespace`) and using this namespace explicitly(`-n custom_ns`) and delete it manually?

Reply:
OK, thanks!